### PR TITLE
BUGFIX: SD-160 handle to show full-height column/bar without min/max setting

### PR DESCRIPTION
**Pull request by naibinh**

### CODE REVIEW

**Comment by naibinh** [comment]
extended test - storybook

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/2470/ : SUCCESS in 10s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/1869/ : SUCCESS in 4m 05s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/2472/ : SUCCESS in 16s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/1870/ : SUCCESS in 4m 36s

**Comment by naibinh** [comment]
extended test - storybook

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/2473/ : SUCCESS in 9s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/1871/ : SUCCESS in 2m 31s

**Comment by yenkins** [comment]
Link to Storybook test result:

https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-storybook-zuul-docker/988/artifact/test-storybook/html-report/index.html

**Comment by gdgate** [comment]
Build succeeded (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/2474/ : SUCCESS in 6s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-pull-request-dispatcher/580/ : SUCCESS in 22m 16s

File: src/components/visualizations/chart/highcharts/plugins/zeroAlignYAxis.ts [hunk 1]
```diff
@@ -155,32 +127,69 @@ function minmaxCanon(minmax: IMinMaxInfo[]): ICanon[] {
     return canon;
 }
 
+function getMinMaxLookup(minmax: IMinMaxInfo[]): IMinMaxLookup {
```
Comment (ivan-nejezchleb) [inline src/components/visualizations/chart/highcharts/plugins/zeroAlignYAxis.ts]: add return types to all new functions

File: src/components/visualizations/chart/highcharts/plugins/zeroAlignYAxis.ts [hunk 2]
```diff
@@ -155,32 +127,69 @@ function minmaxCanon(minmax: IMinMaxInfo[]): ICanon[] {
...
  * @param minmaxLookup
  * @param axisIndex
  */
-function getNewMin(minmax: IMinMaxInfo[], minmaxLookup: IMinMaxLookup, axisIndex: number) {
-    const idx = (minmax[0].min <= minmax[1].min) ? 0 : 1;
-    const f = !minmax[idx].max ?
-                    minmax[idx].min : // handle divide zero case
-                    minmax[idx].min / minmax[idx].max;
-    return f * minmaxLookup[axisIndex].max;
+function getMin(minmax: IMinMaxInfo[], minmaxLookup: IMinMaxLookup, axisIndex: number) {
```
Comment (ivan-nejezchleb) [inline src/components/visualizations/chart/highcharts/plugins/zeroAlignYAxis.ts]: is not possible to rewrite this two functions into one and different parts provide as parameters? Something like this. I am curious about readability and length of such solution

function getLimit(minmax: IMinMaxInfo[], minmaxLookup: IMinMaxLookup, axisIndex: number), getIndex: function, calculateLimit: function, findExtreme: function){
const isMinMaxConfig = isMinMaxConfigOnAnyAxis(minmax);
if (isMinMaxConfig) {
const idx = getIndex(minmax);
return calculateLimit(idx, minmax, minmaxLookup, axisIndex);
}
return findExtreme([0, 1]
.map((index: number) => calculateLimit(index, minmax, minmaxLookup, axisIndex)))

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/2480/ : SUCCESS in 13s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/1880/ : SUCCESS in 7m 34s

**Comment by naibinh** [comment]
extended test - storybook

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/2481/ : SUCCESS in 10s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/1881/ : SUCCESS in 6m 53s

**Comment by yenkins** [comment]
Link to Storybook test result:

https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-storybook-zuul-docker/994/artifact/test-storybook/html-report/index.html

**Comment by gdgate** [comment]
Build failed (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/2482/ : SUCCESS in 9s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-pull-request-dispatcher/582/ : UNSTABLE in 17m 02s

**Comment by naibinh** [comment]
extended test - storybook

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/2483/ : SUCCESS in 10s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/1882/ : SUCCESS in 7m 41s

**Comment by yenkins** [comment]
Link to Storybook test result:

https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-storybook-zuul-docker/995/artifact/test-storybook/html-report/index.html

**Comment by gdgate** [comment]
Build succeeded (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/2484/ : SUCCESS in 9s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-pull-request-dispatcher/583/ : SUCCESS in 18m 53s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/2487/ : SUCCESS in 6s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/1885/ : SUCCESS in 1m 59s

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-merge-compliance-check/530/ : SUCCESS in 19s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/1888/ : SUCCESS in 2m 34s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-storybook-zuul-docker/998/ : SUCCESS in 17m 26s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-examples-zuul-docker/488/ : SUCCESS in 18m 32s

**Pull request merged**